src/System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -15,6 +15,9 @@ public class MockFileSystem : FileSystemBase, IMockFileDataAccessor
 
         private readonly IDictionary<string, FileSystemEntry> files;
         private readonly PathVerifier pathVerifier;
+        [NonSerialized]
+        private Func<DateTime> dateTimeProvider = defaultDateTimeProvider;
+        private static Func<DateTime> defaultDateTimeProvider = () => DateTime.Now;
```

Comment:
Is it really `DateTime.Now` (vs. `UtcNow`) for the real file system?

Reply:
It is stored as local time in [MockFileInfo.cs](https://github.com/TestableIO/System.IO.Abstractions/blob/main/src/System.IO.Abstractions.TestingHelpers/MockFileInfo.cs), e.g.:

```csharp
        /// <inheritdoc />
        public override DateTime CreationTime
        {
            get
            {
                var mockFileData = GetMockFileDataForRead();
                return mockFileData.CreationTime.DateTime;
            }
            set
            {
                var mockFileData = GetMockFileDataForWrite();
                mockFileData.CreationTime = value;
            }
        }
```

Reply:
I adjusted the code to consistently store it in UTC.

During this change I noticed, that some existing unit tests were wrong with regards to the `DateTimeKind` (e.g. in [MockFileInfo_LastWriteTimeUtc_ShouldSetLastWriteTimeUtcOfFileInMemoryFileSystem](https://github.com/vbreuss/System.IO.Abstractions/blob/f1f3eda8477e3a2b5a969cca17b5435065aef4f8/tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs#L434) `LastWriteTime` instead of `LastWriteTimeUtc` is read and written)

Reply:
I think this change might make #879 obsolete?
Or rather the behaviour is, that `DateTimeKind.Unspecified` is now consistently interpreted as local time...